components/business-adaptors/sap/src/main/java/org/wso2/carbon/transports/sap/SAPTransportSender.java

```diff
@@ -140,23 +140,35 @@ public void sendMessage(MessageContext messageContext, String targetEPR,
 
         try {
             URI uri = new URI(targetEPR);
+            if (log.isDebugEnabled()) {
+                log.debug("Started sending message to uri=" + uri);
+            }
             String destName = uri.getPath().substring(1);
             JCoDestination destination = JCoDestinationManager.getDestination(destName);
-
+            if (log.isDebugEnabled()) {
+                log.debug("Retrieved destination: " + destination.getDestinationID());
+            }
             if (uri.getScheme().equals(SAPConstants.SAP_IDOC_PROTOCOL_NAME)) {
                 IDocRepository iDocRepository = JCoIDoc.getIDocRepository(destination);
                 String tid = destination.createTID();
-                IDocDocumentList iDocList = getIDocs(messageContext, iDocRepository);
-
+                if (log.isDebugEnabled()) {
+                    log.debug("Created transaction ID: " + tid);
+                }
                 //Set the transaction id as a transport header so that it can be used later.
                 Object headers = messageContext.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
                 Map headersMap = (Map) headers;
                 headersMap.put(SAP_TRANSACTION_ID, tid);
 
+                IDocDocumentList iDocList = getIDocs(messageContext, iDocRepository);
                 JCoIDoc.send(iDocList, getIDocVersion(uri), destination, tid);
                 destination.confirmTID(tid);
-
+                if (log.isDebugEnabled()) {
+                    log.debug("Successfully sent Idoc");
```

Comment:
Shall we make the log more informative? 
idoc identifier? 
sent to where, through what connection?

Reply:
Added the destination identifier.